operator/pkg/reconciliation/reconcile_reaper.go

```diff
@@ -106,7 +145,16 @@ func (rc *ReconciliationContext) CheckReaperSchemaInitialized() result.Reconcile
 	} else if err != nil {
 		return result.Error(err)
 	} else if jobFinished(schemaJob) {
-		return result.Continue()
+		patch := client.MergeFrom(rc.Datacenter.DeepCopy())
+		rc.Datacenter.Status.ReaperStatus.SchemaInitialized = true
+		if err = rc.Client.Status().Patch(rc.Ctx, rc.Datacenter, patch); err != nil {
+			rc.ReqLogger.Error(err, "error updating the reaper status")
+			return result.Error(err)
+		}
+		// Requeue with a delay to give a chance for C* schema changes to propagate
+		//
+		// TODO Should the delay be adjusted based on the C* cluster size?
+		return result.RequeueSoon(5)
```

Comment:
Is there any way to ask Cassandra if the changes are done propagating? Like a query we can run or something?

Reply:
There is. C* driver provide an API to check for schema agreement. With the Java driver for example, it is just a single method call, `checkSchemaAgreement()`, that returns a boolean. Internally, the driver executes the following queries:

```
SELECT peer, rpc_address, schema_version FROM system.peers
```
and

```
SELECT schema_version FROM system.local WHERE key='local'
```
It checks that `schema_version` is the same across all rows.

It would be easy to just add the `checkSchemaAgreement` call to the schema init job; however, I plan to get rid of the schema init job now that the management api supports creating keyspaces. Given that I will implement a `checkSchemaAgreement` function in the operator.